ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantState.scala

```diff
@@ -23,44 +23,32 @@ import com.daml.ledger.participant.state.v2.{
 import com.daml.lf.data.{Ref, Time}
 import com.daml.lf.transaction.SubmittedTransaction
 import com.daml.logging.LoggingContext
-import com.daml.metrics.Metrics
 import com.daml.telemetry.TelemetryContext
 
 /** Implements read and write operations required for running a participant server.
   *
-  * Adapts [[LedgerReader]] and [[LedgerWriter]] interfaces to [[com.daml.ledger.participant.state.v2.ReadService]] and
-  * [[com.daml.ledger.participant.state.v2.WriteService]], respectively.
+  * Unifies [[KeyValueParticipantStateReader]] and [[KeyValueParticipantStateWriter]] interfaces
   * Will report [[com.daml.ledger.api.health.Healthy]] as health status only if both
   * `reader` and `writer` are healthy.
   *
-  * @param reader       [[LedgerReader]] instance to adapt
-  * @param writer       [[LedgerWriter]] instance to adapt
-  * @param metrics      used to record timing metrics for [[LedgerWriter]] calls
+  * @param reader       [[LedgerReader]] instance to delegate to
+  * @param writer       [[LedgerWriter]] instance to delegate to
   */
 class KeyValueParticipantState(
-    reader: LedgerReader,
-    writer: LedgerWriter,
-    metrics: Metrics,
-    enableSelfServiceErrorCodes: Boolean,
+    reader: KeyValueParticipantStateReader,
```

Comment:
Could `reader` and `writer` be of type `ReadService` and `WriteService`, respectively, instead of a type that implements the latter?